### PR TITLE
Request payment stack

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: app/migration/settings/billing/page.tsx [hunk 1]
```diff
@@ -27,6 +32,7 @@ export default function BillingPage() {
           id={billingProfile?.id}
         />
       </ProfileCard>
+      <Button onClick={open}>Open request payment</Button>
```
Comment (haydencleary) [inline app/migration/settings/billing/page.tsx:35]: Missing translation

File: components/ds/modals/bottom-sheet/bottom-sheet.tsx [hunk 2]
```diff
@@ -37,16 +46,22 @@ export function BottomSheet({ children, onOpen, onClose, title, open = false }:
         },
       }}
     >
-      <ModalContent className="mx-0 my-0 max-w-full rounded-b-none bg-greyscale-900 sm:mx-0 sm:my-0">
+      <ModalContent
+        className={cn("mx-0 my-0 max-w-full rounded-b-none bg-greyscale-900 sm:mx-0 sm:my-0", {
```
Comment (haydencleary) [inline components/ds/modals/bottom-sheet/bottom-sheet.tsx:50]:
```suggestion
        className={cn("m-0 max-w-full rounded-b-none bg-greyscale-900", {
```

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/generate-invoice/generate-invoice.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            <AmountCounter total={1000} />
+            <div className="flex h-auto w-full items-center justify-between gap-5 border-t border-card-border-light bg-card-background-light px-8 py-6">
+              {/* // empty div to keep the flex layout */}
+              {isLoading ? <Spinner /> : <div />}
+              <div className="flex items-center justify-end gap-5 ">
+                <Button variant="secondary" size="m" onClick={() => goTo({ to: TRequestPaymentsStacks.Views.Select })}>
+                  <Translate token="v2.pages.stacks.request_payments.form.back" />
+                </Button>
+                <Button variant="primary" size="m" onClick={() => null}>
+                  Send invoice
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/generate-invoice/generate-invoice.tsx:32]: Missing translation

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/generate-invoice/generate-invoice.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import { Button } from "components/ds/button/button";
+import { AmountCounter } from "components/features/stacks/payments-flow/request-payments-stacks/components/amount-counter/amount-counter";
+import { TRequestPaymentsStacks } from "components/features/stacks/payments-flow/request-payments-stacks/request-payments-stacks.types";
+import { ScrollView } from "components/layout/pages/scroll-view/scroll-view";
+import { Translate } from "components/layout/translate/translate";
+
+import { TGenerateInvoice } from "./generate-invoice.types";
+
+export function GenerateInvoice({ rewardIds, goTo }: TGenerateInvoice.Props) {
+  const isLoading = false;
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/generate-invoice/generate-invoice.tsx:12]: Shouldn't this be state ?

File: src/translations/v2/en/pages/stacks-request-payments.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "title": "Request payment",
+  "form": {
+    "submit": {
+      "zero": "Select __count__ reward",
```
Comment (haydencleary) [inline src/translations/v2/en/pages/stacks-request-payments.json:5]:
```suggestion
      "zero": "Select __count__ rewards",
```

File: src/translations/v2/en/pages/stacks-request-payments.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      "one": "Select __count__ reward",
+      "many": "Select __count__ rewards"
+    },
+    "back": "Back"
+  },
+  "item": {
+    "to": "to",
+    "contributions": {
+      "zero": "__count__ contribution",
```
Comment (haydencleary) [inline src/translations/v2/en/pages/stacks-request-payments.json:14]:
```suggestion
      "zero": "__count__ contributions",
```

File: src/translations/v2/en/pages/stacks-request-payments.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      "one": "__count__ contribution",
+      "many": "__count__ contributions"
+    }
+  },
+  "amount": {
+    "label_individual": "As an individual, you can’t exceed a total of 5,000 USD reward per year.",
```
Comment (haydencleary) [inline src/translations/v2/en/pages/stacks-request-payments.json:20]:
```suggestion
    "label_individual": "As an individual, you can’t exceed a total of 5,000 USD in rewards per year.",
```

File: app/migration/settings/billing/page.tsx (hunk 1 above)
Comment (alexbeno) [reply app/migration/settings/billing/page.tsx:35]: only for testing !

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/generate-invoice/generate-invoice.tsx (hunk 3 above)
Comment (alexbeno) [reply components/features/stacks/payments-flow/request-payments-stacks/features/views/generate-invoice/generate-invoice.tsx:32]: it's not a final screen i have juste create the file for @pixelfact

**Review by haydencleary — APPROVED** [review]

**Pull request merged**